### PR TITLE
fix: browser forwarder listen ip is empty make v2core crash 

### PR DESCRIPTION
**Pull request by cpdyj**

### CODE REVIEW

**Comment by dyhkwong** [comment]
this is a upstream bug? https://github.com/v2fly/v2ray-core/commit/b68411f460215728230bcf988404031420d77443

**Comment by DuckSoft** [comment]
it's good not to set them if the values are empty.

File: src/ui/widgets/editors/w_RoutesEditor.cpp [hunk 1]
```diff
@@ -299,9 +299,14 @@ CONFIGROOT RouteEditor::OpenEditor()
     {
         // Process Browser Forwarder
         QJsonObject browserForwarder;
-        browserForwarder["listenAddr"] = bfListenIPTxt->text();
-        browserForwarder["listenPort"] = bfListenPortTxt->value();
-        root["browserForwarder"] = browserForwarder;
+		if(bfListenIPTxt->text().trimmed() == ""){
```
Comment (DuckSoft) [inline src/ui/widgets/editors/w_RoutesEditor.cpp:302]: I remembered that there is a method called isEmpty on QString...

File: src/ui/widgets/editors/w_RoutesEditor.cpp [hunk 2]
```diff
@@ -299,9 +299,14 @@ CONFIGROOT RouteEditor::OpenEditor()
...
+			root["browserForwarder"] = QJsonValue::Null;
```
Comment (DuckSoft) [inline src/ui/widgets/editors/w_RoutesEditor.cpp:303]: What if we don't set it at all?

**Review by DuckSoft — COMMENTED** [review]
Looks good. Need more polishing before getting merged.
Also remember to reformat the code.

File: src/ui/widgets/editors/w_RoutesEditor.cpp (hunk 2 above)
Comment (cpdyj) [reply src/ui/widgets/editors/w_RoutesEditor.cpp:303]: In fact I don't know what will happen. So I set it null.

File: src/ui/widgets/editors/w_RoutesEditor.cpp (hunk 1 above)
Comment (cpdyj) [reply src/ui/widgets/editors/w_RoutesEditor.cpp:302]: 😅😅😅

**Comment by cpdyj** [comment]
> this is a upstream bug? [v2fly/v2ray-core@b68411f](https://github.com/v2fly/v2ray-core/commit/b68411f460215728230bcf988404031420d77443)

 In fact I don't know it should be an upstream bug or not. I don't agree that empty ip field is a good choice.

**Comment by DuckSoft** [comment]
@cpdyj It's a good and valid Pull Request. I've done the modifications.

**Pull request merged**